src/tool/hpcrun/gpu/gpu-trace-channel.h

```diff
@@ -53,14 +56,19 @@
 #include "gpu-trace.h"
 #include "gpu-trace-item.h"
 
-
-
 //******************************************************************************
 // type declarations
 //******************************************************************************
 
 typedef struct gpu_trace_channel_t gpu_trace_channel_t;
 
+typedef struct gpu_trace_channel_t {
```

Comment:
the problem is that that I am using **gpu_trace_channel_t** data structure in :
gpu-trace.c
gpu-trace-channel-set.c

and if I don't expose the abstraction in header file, it will not be included in these two files and I am getting :
error: dereferencing pointer to incomplete type ‘gpu_trace_channel_t {aka struct gpu_trace_channel_t}’

Is there a better way for this?

Reply:
You can declare `gpu_trace_channel_t` in `gpu-trace-channel.c` or `gpu-trace-channel-set.c`, using `typedef` to expose it, and implement interfaces to extract its members.